bundledatasrc_test.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2019 Canonical Ltd.
+// Licensed under the LGPLv3, see LICENCE file for details.
+
+package charm
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/juju/testing"
+	gc "gopkg.in/check.v1"
+)
+
+type BundleDataSourceSuite struct {
+	testing.IsolationSuite
+}
+
+var _ = gc.Suite(&BundleDataSourceSuite{})
+
+func (s *BundleDataSourceSuite) TestReadMultiDocBundle(c *gc.C) {
+	path := bundleDirPath(c, "wordpress-multidoc")
+	ds := FileBundleDataSource(filepath.Join(path, "bundle.yaml"))
```

Comment:
the decoupling is ok, but it does make it quite a bit harder to figure out what this test is actually doing. The bundle.yaml doesn't look that big, is it reasonable to change this into:

const MultiDocBundle = `
...
`[1:]

...
  ds := StreamBundleDataSource(bytes.Buffer(MultiDocBundle))

?

Reply:
Sure thing. This particular test exercises the FileBundleDataSource implementation. I will update this together with my changes for making the FileBundleDataSource work with local paths and/or archives (see comment above)